File: ChartsDemo/Classes/RealmBase/RealmDemoData.m
//
//  RealmDemoData.m
//  ChartsDemo
//  Copyright © 2015 dcg. All rights reserved.
//

#import "RealmDemoData.h"

@implementation RealmDemoData

- (id)initWithYValue:(double)yValue
{
    self = [super init];
    
    if (self)
    {
        self.yValue = yValue;
    }
    
    return self;
}

- (id)initWithXValue:(double)xValue
              yValue:(double)yValue
{
    self = [super init];
    
    if (self)
    {
        self.xValue = xValue;
        self.yValue = yValue;
    }
    
    return self;
}

- (id)initWithXValue:(double)xValue
         stackValues:(NSArray<NSNumber *> *)stackValues
{
    self = [super init];
    
    if (self)
    {
        self.xValue = xValue;
        self.stackValues = [[RLMArray<RealmFloat> alloc] initWithObjectClassName:@"RealmFloat"];
        
        for (NSNumber *value in stackValues)
        {
            [self.stackValues addObject:[[RealmFloat alloc] initWithFloatValue:value.floatValue]];
        }
    }
    
    return self;
}

- (id)initWithXValue:(double)xValue
                high:(double)high
                 low:(double)low
                open:(double)open
               close:(double)close
{
    self = [super init];
    
    if (self)
    {
        self.xValue = xValue;
        self.yValue = (high + low) / 2.f;
        self.high = high;
        self.low = low;
        self.open = open;
        self.close = close;
    }
    
    return self;
}

- (id)initWithXValue:(double)xValue
              yValue:(double)yValue
          bubbleSize:(double)bubbleSize
{
    self = [super init];
    
    if (self)
    {
        self.xValue = xValue;
        self.yValue = yValue;
        self.bubbleSize = bubbleSize;
    }
    
    return self;
}

- (id)initWithYValue:(double)yValue
               label:(NSString *)label
{
    self = [super init];
    
    if (self)
    {
        self.yValue = yValue;
        self.label = label;
    }
    
    return self;
}

@end

File: ChartsDemo-OSX/ChartsDemo-OSX/RealmBase/RealmDemoData.swift
//
//  RealmDemoData.swift
//  ChartsDemo-OSX
//
//  Created by thierryH24A on 02/04/2017.
//  Copyright © 2017 dcg. All rights reserved.
//

import Cocoa
import Realm

class RealmDemoData: RLMObject
{
    dynamic var xValue = 0.0
    dynamic var yValue = 0.0
    
    dynamic var open = 0.0
    dynamic var close = 0.0
    dynamic var high = 0.0
    dynamic var low = 0.0
    
    dynamic var bubbleSize = 0.0

    dynamic var stackValues = RLMArray(objectClassName: "RealmFloat")
    
    dynamic var label = ""
    dynamic var someStringField = ""
    
    override init() {
        super.init()
    }
    
    init(yValue: Double) {
        super.init()
        
        self.yValue = yValue
    }
    
    init(xValue: Double, yValue: Double) {
        super.init()
        
        self.xValue = xValue
        self.yValue = yValue
    }
    
    init(xValue: Double, stackValues: [NSNumber]) {
        super.init()
        
        self.xValue = xValue
        
        for value: NSNumber in stackValues
        {
            self.stackValues.add(RealmFloat(floatValue: CFloat(value)))
        }
    }
    
    init(xValue: Double, high: Double, low: Double, open: Double, close: Double)
    {
        super.init()
        
        self.xValue = xValue
        yValue = (high + low) / 2.0
        self.high = high
        self.low = low
        self.open = open
        self.close = close
    }
    
    init(xValue: Double, yValue: Double, bubbleSize: Double) {
        super.init()
        
        self.xValue = xValue
        self.yValue = yValue
        self.bubbleSize = bubbleSize
    }

    init(yValue: Double, label: String) {
        super.init()
        
        self.yValue = yValue
        self.label = label
    }
}
